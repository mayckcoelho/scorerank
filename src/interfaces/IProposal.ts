interface IProposal {
    id: string;
    image: string;
    tax: string;
    accepted: boolean;
    value: string;
}

export default IProposal;