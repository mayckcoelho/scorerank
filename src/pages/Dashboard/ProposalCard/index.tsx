import React, { useState } from 'react';
import IProposal from "../../../interfaces/IProposal";
import CardDetail from "../../../components/CardDetail";
import Modal from "../../../components/Modal";
import { useAuth } from '../../../contexts/auth';
import { useDispatch } from 'react-redux'
import { accept } from "../../../ducks/proposals";

interface Props {
    proposal: IProposal;
}

const ProposalCard: React.FC<Props> = ({ proposal }) => {
    const [visible, setVisible] = useState(false)
    const { updateScore } = useAuth();
    const dispatch = useDispatch();

    return (
        <>
            <CardDetail key={proposal.id}
                onPress={() => setVisible(true)}
                icon="credit-card"
                color="#2c585f"
                title="Propostas de crédito"
                description="Encontramos ofertas de cartão de crédito!"
                image={proposal.image}
                sub={proposal.tax}
                imageDescription={`Limite de ${proposal.value}`}
                bottom="Ver oferta" />
            <Modal
                isVisible={visible}
                image={proposal.image}
                close={() => setVisible(false)}
                title="Aproveite esta oferta de crédito e aumente seu Score!"
                sub={proposal.tax}
                description={`Limite de ${proposal.value}`}
                action={() => {
                    dispatch(accept(proposal.id));
                    setVisible(false);
                    updateScore();
                }}
                actionText="Aceitar proposta"
            />
        </>
    )
}

export default ProposalCard;