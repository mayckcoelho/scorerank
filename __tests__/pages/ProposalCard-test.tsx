import React from 'react';
import ProposalCard from '../../src/pages/Dashboard/ProposalCard';
import IProposal from "../../src/interfaces/IProposal"
import renderer from 'react-test-renderer';

describe('ProposalCard should render correctly', () => {
    it('match snapshot', () => {
        const proposalcard = renderer
            .create(<ProposalCard proposal={{} as IProposal} />)
            .toJSON();
        expect(proposalcard).toMatchSnapshot();
    });
})