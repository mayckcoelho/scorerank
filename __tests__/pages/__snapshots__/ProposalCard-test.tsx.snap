// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ProposalCard should render correctly match snapshot 1`] = `
Array [
  <View
    style={
      Array [
        Object {
          "borderRadius": 6,
          "marginBottom": 20,
          "marginLeft": 0,
          "marginRight": 0,
          "marginTop": 20,
          "paddingBottom": 20,
          "paddingLeft": 20,
          "paddingRight": 20,
          "paddingTop": 20,
          "shadowColor": "rgba(0, 0, 0, .3)",
          "shadowOffset": Object {
            "height": 5,
            "width": 0,
          },
          "shadowOpacity": 1,
          "shadowRadius": 8,
          "width": "95%",
        },
        Object {
          "backgroundColor": "#fff",
        },
      ]
    }
  >
    <View
      style={
        Array [
          Object {
            "alignItems": "center",
            "flexDirection": "row",
          },
        ]
      }
    >
      <Text
        allowFontScaling={false}
        style={
          Array [
            Object {
              "color": "#2c585f",
              "fontSize": 22,
            },
            undefined,
            Object {
              "fontFamily": "FontAwesome",
              "fontStyle": "normal",
              "fontWeight": "normal",
            },
            Object {},
          ]
        }
      >
        
      </Text>
      <Text
        style={
          Array [
            Object {
              "fontSize": 18,
              "marginLeft": 10,
            },
            Object {
              "color": "#2c585f",
            },
          ]
        }
      >
        Propostas de crédito
      </Text>
    </View>
    <View
      style={
        Array [
          Object {
            "backgroundColor": "#676767",
            "height": 1,
            "marginBottom": 20,
            "marginLeft": 0,
            "marginRight": 0,
            "marginTop": 20,
            "width": "100%",
          },
        ]
      }
    />
    <Text
      style={
        Array [
          Object {
            "color": "#676767",
            "fontSize": 18,
          },
        ]
      }
    >
      Encontramos ofertas de cartão de crédito!
    </Text>
    <View
      style={
        Array [
          Object {
            "alignItems": "center",
            "flexDirection": "row",
            "justifyContent": "space-between",
            "marginBottom": 30,
            "marginLeft": 0,
            "marginRight": 0,
            "marginTop": 30,
          },
        ]
      }
    >
      <Image
        resizeMode="contain"
        source={
          Object {
            "uri": undefined,
          }
        }
        style={
          Array [
            Object {
              "height": 70,
              "width": 120,
            },
          ]
        }
      />
      <View>
        <Text
          style={
            Array [
              Object {
                "alignSelf": "flex-end",
                "color": "#b2b2b2",
                "fontSize": 16,
              },
              Object {
                "textDecorationLine": "none",
              },
            ]
          }
        />
        <Text
          style={
            Array [
              Object {
                "alignSelf": "flex-end",
                "color": "#757575",
                "fontSize": 20,
                "marginBottom": 5,
                "marginLeft": 0,
                "marginRight": 0,
                "marginTop": 5,
              },
            ]
          }
        >
          Limite de undefined
        </Text>
      </View>
    </View>
    <View
      accessible={true}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "opacity": 1,
        }
      }
    >
      <Text
        style={
          Array [
            Object {
              "alignSelf": "center",
              "color": "#6a1e55",
              "fontSize": 18,
              "fontWeight": "bold",
              "marginRight": 5,
            },
          ]
        }
      >
        Ver oferta
      </Text>
    </View>
  </View>,
  <Modal
    animationType="none"
    hardwareAccelerated={false}
    hideModalContentWhileAnimating={false}
    onModalHide={[Function]}
    onModalWillHide={[Function]}
    onModalWillShow={[Function]}
    onRequestClose={[Function]}
    scrollHorizontal={false}
    scrollOffset={0}
    scrollOffsetMax={0}
    scrollTo={null}
    supportedOrientations={
      Array [
        "portrait",
        "landscape",
      ]
    }
    swipeThreshold={100}
    transparent={true}
    visible={false}
  >
    <View
      accessible={true}
      focusable={true}
      forwardedRef={[Function]}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "backgroundColor": "black",
          "bottom": 0,
          "height": 1334,
          "left": 0,
          "opacity": 0,
          "position": "absolute",
          "right": 0,
          "top": 0,
          "width": 750,
        }
      }
    />
    <View
      forwardedRef={[Function]}
      hideModalContentWhileAnimating={false}
      onModalHide={[Function]}
      onModalWillHide={[Function]}
      onModalWillShow={[Function]}
      pointerEvents="box-none"
      scrollHorizontal={false}
      scrollOffset={0}
      scrollOffsetMax={0}
      scrollTo={null}
      style={
        Object {
          "flex": 1,
          "justifyContent": "center",
          "margin": 0,
          "transform": Array [
            Object {
              "translateY": 0,
            },
          ],
        }
      }
      supportedOrientations={
        Array [
          "portrait",
          "landscape",
        ]
      }
      swipeThreshold={100}
    >
      <View
        style={
          Array [
            Object {
              "backgroundColor": "#fff",
              "borderTopLeftRadius": 15,
              "borderTopRightRadius": 15,
              "bottom": 0,
              "height": "50%",
              "justifyContent": "space-between",
              "marginBottom": 0,
              "marginLeft": 0,
              "marginRight": 0,
              "marginTop": 0,
              "paddingBottom": 15,
              "paddingLeft": 15,
              "paddingRight": 15,
              "paddingTop": 0,
              "position": "absolute",
              "width": "100%",
            },
          ]
        }
      >
        <View
          style={
            Array [
              Object {
                "alignSelf": "center",
                "backgroundColor": "#ccc",
                "borderRadius": 50,
                "height": 5,
                "marginTop": 5,
                "width": 50,
              },
            ]
          }
        />
        <Image
          resizeMode="contain"
          source={
            Object {
              "uri": undefined,
            }
          }
          style={
            Array [
              Object {
                "alignSelf": "center",
                "height": 70,
                "width": 120,
              },
            ]
          }
        />
        <Text
          style={
            Array [
              Object {
                "alignSelf": "flex-end",
                "color": "#757575",
                "fontSize": 20,
                "marginBottom": 5,
                "marginLeft": 0,
                "marginRight": 0,
                "marginTop": 5,
              },
              Object {
                "alignSelf": "flex-start",
              },
            ]
          }
        >
          Aproveite esta oferta de crédito e aumente seu Score!
        </Text>
        <Text
          style={
            Array [
              Object {
                "alignSelf": "flex-end",
                "color": "#b2b2b2",
                "fontSize": 16,
              },
            ]
          }
        />
        <Text
          style={
            Array [
              Object {
                "alignSelf": "flex-end",
                "color": "#757575",
                "fontSize": 20,
                "marginBottom": 5,
                "marginLeft": 0,
                "marginRight": 0,
                "marginTop": 5,
              },
            ]
          }
        >
          Limite de undefined
        </Text>
        <View
          accessible={true}
          focusable={true}
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "alignItems": "center",
              "backgroundColor": "#ca6a38",
              "borderRadius": 10,
              "height": 50,
              "justifyContent": "center",
              "marginTop": 30,
              "opacity": 1,
              "width": "100%",
            }
          }
        >
          <Text
            style={
              Array [
                Object {
                  "color": "#fff",
                },
              ]
            }
          >
            Aceitar proposta
          </Text>
        </View>
        <View
          accessible={true}
          focusable={true}
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "alignItems": "center",
              "backgroundColor": "#ccc",
              "borderRadius": 10,
              "height": 50,
              "justifyContent": "center",
              "opacity": 1,
              "width": "100%",
            }
          }
        >
          <Text
            style={
              Array [
                Object {
                  "color": "#000",
                },
              ]
            }
          >
            Fechar
          </Text>
        </View>
      </View>
    </View>
  </Modal>,
]
`;
